### PR TITLE
IDEX-3566: Forbid commit if user name and email aren't set

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur ivinokur@codenvy.com

@skabashnyuk @vparfonov please review

### CODE REVIEW

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java [hunk 1]
```diff
@@ -48,8 +43,12 @@ public CommitterPreferencePresenter(CommitterPreferenceView view,
         this.view = view;
         this.preferencesManager = preferencesManager;
 
-        name = firstNonNull(preferencesManager.getValue(COMMITTER_NAME), DEFAULT_COMMITTER_NAME);
-        email = firstNonNull(preferencesManager.getValue(COMMITTER_EMAIL), DEFAULT_COMMITTER_EMAIL);
+        if (preferencesManager.getValue(COMMITTER_NAME) != null ) {
```
Comment (sleshchenko) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java]: Why do you check value on null before storing value into variable which have null value?

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java [hunk 2]
```diff
@@ -97,9 +96,6 @@ public void storeChanges() {
     /** {@inheritDoc} */
     @Override
     public void revertChanges() {
-        name = firstNonNull(preferencesManager.getValue(COMMITTER_NAME), DEFAULT_COMMITTER_NAME);
```
Comment (sleshchenko) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java:100]: I think you should change it to 

```
name = preferencesManager.getValue(COMMITTER_NAME);
email = preferencesManager.getValue(COMMITTER_EMAIL);
```

**Comment by skabashnyuk** [comment]
@mshaposhnik ?

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java (hunk 1 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java]: removed the check on null

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java (hunk 2 above)
Comment (vinokurig) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java:100]: done

**Comment by mshaposhnik** [comment]
Pls refactor LocalGitUserResolver javadoc since it is not looking in environmemt anymore

**Comment by mshaposhnik** [comment]
Maybe we can wait 1-2h to error codes push? So we will be able to avoid relying on exception messages?

File: core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/TagCreateCommand.java [hunk 3]
```diff
@@ -42,6 +42,12 @@ public Tag execute() throws GitException {
         if (name == null) {
             throw new GitException("Name wasn't set.");
         }
+        if (committer == null) {
+            throw new GitException("Committer can't be null");
```
Comment (sleshchenko) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/TagCreateCommand.java:45]: another error messages with dot at the end. Do in the same way

Comment (vinokurig) [inline core/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/commands/TagCreateCommand.java:45]: done

**Comment by vinokurig** [comment]
Reworked according to error codes, removed git status when commit failed @mshaposhnik, @vparfonov @skabashnyuk please review

**Comment by mshaposhnik** [comment]
Ok

**Comment by vparfonov** [comment]
ok

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**